subsys/bluetooth/mesh/adv_legacy.c

```diff
@@ -35,10 +35,11 @@ LOG_MODULE_REGISTER(bt_mesh_adv_legacy);
 #define ADV_INT_DEFAULT_MS 100
 #define ADV_INT_FAST_MS    20
 
+#define ADV_ENABLED (!atomic_test_bit(bt_mesh.flags, BT_MESH_SUSPENDED))
```

Comment:
I'd suggest to add an input parameter to this macro (either `bt_mesh` or `bt_mesh.flags`). Without it you get the impression that this macro evaluates to a build-time constant, which is not true.